include/fiction/networks/views/input_ordering_view.hpp

```diff
@@ -0,0 +1,510 @@
+//
+// Created by benjamin on 11.04.23.
+//
+
+#ifndef FICTION_INPUT_ORDERING_VIEW_HPP
+#define FICTION_INPUT_ORDERING_VIEW_HPP
+
+#include "fiction/utils/network_utils.hpp"
+
+#include <mockturtle/networks/detail/foreach.hpp>
+#include <mockturtle/traits.hpp>
+#include <mockturtle/views/fanout_view.hpp>
+#include <mockturtle/views/immutable_view.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <iostream>
+#include <optional>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * A variation of mockturtle::topo_view that computes the topological order after sorting the inputs according to the
+ * connections of their fanouts. This is part of the Distribution Newtork I: Ordering_Network
+ *
+ * @tparam Ntk mockturtle network type.
+ * @tparam sorted Flag that determines whether Ntk is already wrapped in a topo_view.
+ */
+template <class Ntk, bool sorted = mockturtle::is_topologically_sorted_v<Ntk>>
```

Comment:
You're checking for the existence of a topological sorting here, but the `input_ordering_view` does not set it itself. Can you briefly explain the reasoning behind it? The way it is implemented, it is possible to layer multiple `input_ordering_views` on top of each other and the ordering would be recomputed every time. That does not seem intended, or is it?